src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -1061,11 +1062,20 @@ public void OnSearchElementClicked(NodeModel nodeModel, Point position)
 
         internal void OnRequestConnectToPort(string nodeCreationName, PortModel portModel)
         {
+            // Do not auto connect code block node since default code block node do not have output port
             if (!nodeCreationName.Contains("Code Block"))
             {
+                //dynamoViewModel.ExecuteCommand(new DynamoModel.MakeConnectionCommand(nodeModel.GUID,))
+                var id = Guid.NewGuid();
                 // Create a new node based on node creation name and connect ports
-                dynamoViewModel.ExecuteCommand(new DynamoModel.CreateAndConnectNodeCommand(Guid.NewGuid(), portModel.Owner.GUID,
-                    nodeCreationName, 0, portModel.Index, portModel.CenterX - 50, portModel.CenterY + 200, false, false));
+                var initialNode = dynamoViewModel.CurrentSpaceViewModel.Nodes.Where(x => x.Id == portModel.Owner.GUID).FirstOrDefault();
+                // Placing the new node based on which input port it is connecting to.
+                // The larger index for the input port, the lower the new node will be placed. 60 and 200 and offset we can adjust.
+                double adjustedY = initialNode.Y + (portModel.Index - 1) * 60 ;
+                // Placing the new node to the left of initial node
+                var adjustedX = initialNode.X - 200; 
+                dynamoViewModel.ExecuteCommand(new DynamoModel.CreateAndConnectNodeCommand(id, portModel.Owner.GUID,
+                    nodeCreationName, 0, portModel.Index, adjustedX, adjustedY, false, false));
```

Comment:
Can we split this step into a CreateNodeCommand step and then add a new command called something like ``PlaceAndConnectCommand``? That way we can first create the new node, get its height and width and use the height and width to place it relative to the original node. In other words, the `PlaceAndConnectCommand`
will be a combination of 2 steps: 1) place the node based on a calculated X and Y position, and 2) call the
`MakeConnectionCommand`. This way we might be able to do away with these hard-coded values.

Reply:
@aparajit-pratap I did thought about that before implementation but the height and width for the new nodes are always default to 100 for node model..

Reply:
![image](https://user-images.githubusercontent.com/3942418/95620793-b4a69c00-0a3e-11eb-9f94-0459ed1e2238.png)
@aparajit-pratap This is what I am referring to. I already considered this solution before finalizing the PR. For now there isn't a way I can see we can get the actual height and width for the newly placed node.

Reply:
Another slightly less important reason is about undo/redo, the current implementation only requires one undo to get back to the previous state versus the split way will need to be done by two undos.

Reply:
I'm trying something else, will get back to you on this PR, I may commit directly to this PR if I can get something working.

Reply:
Couldn't get it to work. In any case this is the problem with this approach right now:
![create_and_connect_nodes](https://user-images.githubusercontent.com/5710686/95625375-672e2d00-0a46-11eb-921c-32d07be193bf.gif)

Reply:
Indeed, placing any slider or any giant node will result in such problem but I dont see a way to be smarter yet..